components/salsa-macros/src/salsa_struct.rs

```diff
@@ -414,6 +414,36 @@ where
         self.args.debug.is_some()
     }
 
+    pub fn serializable(&self) -> bool {
+        self.args.serde.is_some()
+    }
+
+    /// Returns the path to the `serialize` and `deserialize` functions as an optional iterator.
+    pub(crate) fn serde_fn(
+        &self,
+    ) -> syn::Result<impl Iterator<Item = proc_macro2::TokenStream> + '_> {
+        let path = match &self.args.serde {
+            None => None,
+            Some(SerializeOptions::Custom {
+                serialize_fn: Some(serialize_fn),
+                deserialize_fn: Some(deserialize_fn),
+            }) => Some(quote! { (#serialize_fn, #deserialize_fn) }),
+            // Fallback to the `{Serialize, Deserialize}` implementations tuples if only one of the
+            // two is provided.
```

Comment:
There is a little bit of weirdness here in that all of the following are accepted:
```rust
#[salsa::tracked(serialize)]
#[salsa::tracked(deserialize)]
#[salsa::tracked(serialize, deserialize)]
#[salsa::tracked(serialize = ...)]
#[salsa::tracked(deserialize = ...)]
#[salsa::tracked(serialize = ..., deserialize = ...)]
struct ...;
```
But I'm not sure disallowing any of those is better.

Reply:
I moved this into a `persist` attribute to make things more consistent:
```rust
#[salsa::tracked(persist)]
#[salsa::tracked(persist(serialize = ...))]
#[salsa::tracked(persist(deserialize = ...))]
#[salsa::tracked(persist(serialize = ..., deserialize = ...))]
struct ...;
```

We don't accept the `serialize` and `deserialize` arguments on tracked functions because both the input and output types need to implement `serde::{Serialize, Deserialize}`, and the arguments would be ambiguous.